### PR TITLE
Several improvements, if ya'll are interested

### PR DESCRIPTION
**Pull request by murrple-1**
Removing Mandrill and SendGrid, as they weren't implemented and didn't look like they would be.

Adding ability to send email body as one of the fax attachments.

Adding security features: Whitelisting recipients/senders, regex to ensure body is safe, ensuring signature of Mailgun is correct.

Adding Environment variables: MAILGUN_KEY. RECIPIENT_WHITELIST_FILE, SENDER_WHITELIST_FILE, BODY_REGEX.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,26 +3,6 @@ Mailphax
 
 Send faxes with Phaxio using 3rd party email services.  Mailphax is a simple sinatra app.  You can run it on any host or with any service that supports ruby and sinatra.
 
-
-Installation on Heroku
```
Comment (jnankin) [inline README.md:7]: Any reason this is being removed?

File: app.json [hunk 2]
```diff
@@ -42,6 +57,6 @@
       "required": false
     }
   },
-  "repository" : "https://github.com/phaxio/mailphax",
+  "repository" : "https://gitlab.com/tsltd/mailphax.git",
```
Comment (jnankin) [inline app.json]: can't merge like this

File: mailphax.rb [hunk 3]
```diff
@@ -2,62 +2,170 @@
 require 'phaxio'
 require 'mail'
 require 'pony'
+require 'tempfile'
+require 'openssl'
+require 'to_regexp'
 
-if not ENV['PHAXIO_KEY'] or not ENV['PHAXIO_SECRET']
-  raise "You must specify your phaxio API keys in PHAXIO_KEY and PHAXIO_SECRET"
+if not ENV['PHAXIO_KEY'] or not ENV['PHAXIO_SECRET'] or not ENV['MAILGUN_KEY']
```
Comment (jnankin) [inline mailphax.rb:9]: mailgun is not necessarily required for all installations

File: mailphax.rb [hunk 4]
```diff
@@ -2,62 +2,170 @@
...
+get '/mailgun' do
+  [400, "Mailgun supported, but callbacks must be POSTs"]
+end
 
-get '/mandrill' do
-  [501, "mandrill not implemented yet"]
+$recipientWhitelist = nil
+
+def getRecipientWhitelist()
+  if $recipientWhitelist.nil?
```
Comment (jnankin) [inline mailphax.rb:24]: We should document these features in the readme (the new whitelists)

**Review by jnankin — COMMENTED** [review]
I wrote this back when I was a PHP guy, so all the method names are camel case.  I think the ruby convention is snake_case. New code should probably be switched to snake case and not make this worse.

File: mailphax.rb (hunk 3 above)
Comment (jnankin) [reply mailphax.rb:9]: I see now that you addressed this in your comments and removed the other mail providers.  you're probably right.  we don't have the man power right now to implement the other providers unless they're requested.

**Comment by murrple-1** [comment]
I have done goofed. I forgot this pull request existed, and I have been recently working on this branch for some work-project related functionality.

Please, allow me to close this pull request and resubmitted it on a more appropriately named branch, without some of the work-related specifics.

File: README.md (hunk 1 above)
Comment (murrple-1) [reply README.md:7]: Botch pull-request branch. This was intended for strictly a work-related project. Undone in #3.

File: mailphax.rb (hunk 4 above)
Comment (murrple-1) [reply mailphax.rb:24]: I think they are useful, and so I've added to the pull request, but I can also see it being too granular. If it is a worthwhile feature, I can add some documentation.

**Pull request closed without merging**